### PR TITLE
fix(replay): rebuild full snapshot at paused seek boundary

### LINKED ISSUE #4239: replayer: pausing exactly at a FullSnapshot's timestamp skips the rebuild
## Bug

In the replay machine's `play` action, the fast-forward set is `event.timestamp < baselineTime` (strict). An event whose timestamp exactly equals the seek target is classified as *future*, scheduled on the timer, and — on a paused seek — immediately discarded when the PAUSE clears the timer.

When that boundary event is a **FullSnapshot**, the rebuild is skipped entirely: `rebuildFullSnapshot` never runs, so the previously rendered frame stays on screen (and imperative state like the fullscreen `rr_fullscreen` marker leaks across the scrub, since it is only cleared in the FullSnapshot cast).

## Repro

There is a red regression test pinning this down, merged as `it.fails` in `test/replayer.test.ts` (`applies the full snapshot when pausing exactly at its timestamp`, added in #4226): seek to 1500, then seek back to exactly the snapshot's offset — the later frame's DOM survives.

It also surfaced as a rare flake in `test/fullscreen.test.ts`: the test derives `beforeOffset = enterOffset - 1` from a live recording, and when recording timing jitter put the fullscreen-enter event 1ms after the FullSnapshot, `beforeOffset` landed exactly on the snapshot timestamp. That test now pins its offset off the boundary.

## Why the obvious fix is wrong

Flipping the comparison to `<=` moves every boundary event out of the timer path, which breaks `lastPlayedEvent` bookkeeping other flows rely on — `DomContentLoaded`/`Load` events at the recording start are filtered in sync mode (no `CAST_EVENT`), so `play(0)` stops refreshing `lastPlayedEvent`, and a subsequent seek can take the skip-already-applied delta path against a mirror that a `PlayBack` reset has already invalidated. Tried in #4226 and reverted after it broke seven replayer e2e tests deterministically.

A real fix probably needs to either special-case the tie only for the FullSnapshot/paused-seek case, or make `PlayBack` also reset `lastPlayedEvent` so an empty resync can't leave stale delta state behind.

## Done when

The `it.fails` test in `test/replayer.test.ts` is flipped to a passing `it`, and the `beforeOffset` pin in `test/fullscreen.test.ts` can be reverted.

### PR DESCRIPTION
**Pull request by marandaneto**
## Problem

When a paused replay seek lands exactly on a FullSnapshot timestamp, the replay machine classifies the snapshot as a future event. `pause(timeOffset)` then immediately clears the timer, so the snapshot is never rebuilt and the previously rendered frame remains visible. A paired Meta event at the same timestamp is also discarded, which can preserve stale viewport dimensions.

Fixes #4239.

## Changes

- Apply a FullSnapshot and its same-timestamp Meta event synchronously when the snapshot exactly matches the seek baseline.
- Keep the existing strict boundary behavior for all unrelated events to avoid changing `lastPlayedEvent` bookkeeping.
- Add a regression test that seeks forward and then pauses exactly at the FullSnapshot boundary, verifying both the rebuilt DOM and viewport dimensions.
- Add a patch changeset for `posthog-js`.
- Document that modules under `packages/rrweb/` are built with `posthog-js` and do not receive separate changesets.

### Libraries affected

- [ ] All of them
- [x] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [ ] posthog-node
- [ ] posthog-react-native
- [ ] @posthog/react-native-plugin
- [ ] @posthog/react
- [ ] @posthog/ai
- [ ] @posthog/convex
- [ ] @posthog/next
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/openfeature-node-provider
- [ ] @posthog/openfeature-web-provider
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin
- [ ] @posthog/types
- [ ] @posthog/browser-common

## Checklist

- [x] Tests for new code
- [x] Accounted for the impact of any changes across different platforms
- [x] Accounted for backwards compatibility of any changes (no breaking changes!)
- [x] Took care not to unnecessarily increase the bundle size

### If releasing new changes

- [x] Ran `pnpm changeset` to generate a changeset file

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

Implemented with the Pi coding agent after reviewing #4239 and its relationship to #4226. The broader `<` to `<=` change was rejected because it changes boundary handling for every event and had already caused replay bookkeeping regressions; this PR uses the narrower snapshot-boundary behavior and verifies it with the reported seek sequence.

### CODE REVIEW

**Comment by marandaneto** [comment]
Speculative fix and 0 context so @PostHog/team-replay pls double check

File: packages/rrweb/rrweb/src/replay/machine.ts [hunk 1]
```diff
@@ -209,7 +209,14 @@ export function createPlayerService(
...
               continue;
             }
-            if (event.timestamp < baselineTime) {
+            // A paused seek clears future timer actions before this event can run.
+            const isFullSnapshotAtSeekBoundary =
+              event.type === EventType.FullSnapshot &&
+              event.timestamp === baselineTime;
+            if (
+              event.timestamp < baselineTime ||
+              isFullSnapshotAtSeekBoundary
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply packages/rrweb/rrweb/src/replay/machine.ts]: Good catch! Fixed in 2b30a3e36 — same-timestamp Meta events are now applied synchronously with the boundary FullSnapshot, and the regression test verifies that viewport dimensions are restored.

**Review by dustinbyrne — APPROVED** [review]

**Review by arnohillen — APPROVED** [review]
